src/System.Xml.XmlSerializer/src/System/Xml/Serialization/XmlSerializationReaderILGen.cs

```diff
@@ -1097,15 +1097,15 @@ private void WriteEnumMethod(EnumMapping mapping)
                 ilg.Stloc(localTmp);
                 ilg.Ldloc(localTmp);
                 ilg.Brfalse(defaultLabel);
-                InternalHashtable cases = new InternalHashtable();
+                var cases = new HashSet<string>();
                 for (int i = 0; i < constants.Length; i++)
                 {
                     ConstantMapping c = constants[i];
 
                     CodeIdentifier.CheckValidIdentifier(c.Name);
-                    if (cases[c.XmlName] == null)
+                    if (!cases.Contains(c.XmlName))
                     {
-                        cases[c.XmlName] = c.XmlName;
+                        cases.Add(c.XmlName);
```

Comment:
Nit: Same comment about using single call to `HashSet<T>.Add` instead of `Contains` & `Add`.